### PR TITLE
table view fix for custom games Issue #2132

### PR DESCRIPTION
**Pull request by dev-nulled**
First time actually sending code... so let me know where things need to be done differently.  I think this resolves the issues I was seeing with custom games, table view - where you would end up seeing the list go catawampus with duplicates over time.  The min values were null

https://github.com/FAForever/downlords-faf-client/issues/2132

Not sure if zero is a correct floor as a default value.  UI will display >=0 rather than blank for these results too, which may or may not be intended.

### CODE REVIEW

File: src/main/java/com/faforever/client/remote/domain/RatingRange.java [hunk 1]
```diff
@@ -24,7 +24,7 @@ public int compareTo(@NotNull RatingRange o) {
   }
 
   public Integer getMin() {
-    return min;
+    return min == null ? 0 : min;
```
Comment (Brutus5000) [inline src/main/java/com/faforever/client/remote/domain/RatingRange.java:27]: 0 is not the minimum rating. It can be negative as well.

**Comment by Sheikah45** [comment]
What lead you to believe that this is the issue? If it is because of a null value it would be better to do the null handling where needed rather than set a minimum value because there is no minimum value since ratings can be negative as well and this would have the side effect of causing those low rated players to get the warning on every game.

I am unable to reproduce the issue on my end so it is difficult for me to find where it goes wrong.

**Comment by dev-nulled** [comment]
@Sheikah45 I'm also digging into that code as well, but for something where the min is empty, zero seems like a reasonable default.  There is still the possibility of a game where the bottom is less than zero (thus the waiting for a -300 to spin up a game for a few hours) and that still works - as that was not null.  This might have something to do with players who are still in a grey state or never played a ladder game.  Still trouble shooting that.  The root cause was the getMin() function is called on the table render.  The compare checks for null, but the getter does not.  There are a few other places where getMin was goofing up and not having a check - listing on the replay vault also gets horked up over time.  My account (*heelix") is one of those grey accounts, if this is a user issue where it renders fine for some, not for the others.

I linked a few screen shots in the issues.  After an hour or so of the refresh of the table layout, all the games will look like one of the early games.  It works like it use to with that min tweak.  This seemed to have broken just before the default was changed from Custom games to Matchmaking.

I'd actually default the min value to zero on the creation of the RankingRange object in the constructor, and allow it to be set lower... but I wanted to actually talk to someone about this first.  Using an Optional is tempting if we want both a value and a null state.

Without the fix, I see this exception spammed in my logs all day long.

Exception in thread "JavaFX Application Thread" java.lang.NullPointerException: Cannot invoke "java.lang.Integer.intValue()" because the return value of "com.faforever.client.remote.domain.RatingRange.getMin()" is null
	at com.faforever.client.remote.domain.RatingRange.compareTo(RatingRange.java:23)
	at com.faforever.client.remote.domain.RatingRange.compareTo(RatingRange.java:5)
	at javafx.scene.control.TableColumnBase.lambda$static$0(TableColumnBase.java:138)
	at com.sun.javafx.scene.control.TableColumnComparatorBase$TableColumnComparator.doCompare(TableColumnComparatorBase.java:120)
	at com.sun.javafx.scene.control.TableColumn

**Comment by Sheikah45** [comment]
Ah got it so it seems to be an issue when sorting the table by the rating range. the fix would then be to just define a comparator that also accounts for nulls for that column

**Comment by Sheikah45** [comment]
But we still dont want to have a default min value in the rating range as that would have unintended side effects with regards to the display and showing games to players and determining elgibility.

**Comment by Sheikah45** [comment]
Take a look at #2150 which should fix the issue

**Comment by dev-nulled** [comment]
@Sheikah45 The way this is done, it preserves the negative scores if present/set.  The current table view is very unusable right now, if you leave it open for any length of time.

I see some poor soul who is hosting a -500 to 1000 right now.

ranking

**Comment by Sheikah45** [comment]
Yes but if the range is not specified with this implementation then it will default to 0 and all the negative rated players will get warnings that this game is not intended for them which is not the intended behavior. 

No Rating range means no one should get warnings and that there is no intended range.

**Comment by heelix** [comment]
Good point.  I'll pull your branch and see what I see on the table
refreshing.  The label shows >=0 to zero, but not sure if a negative player
actually sees a block.  Will confirm here in a sec.

**Comment by dev-nulled** [comment]
@Sheikah45 Your update to the compare fixes the table layout issue as well as the weirdness I was seeing with the Vault replays.  Going to close mine.

**Pull request closed without merging**